packages/wxr/src/1.3/index.js

```diff
@@ -0,0 +1,665 @@
+	 */
+	async connect( { reset = false } = {} ) {
+		// Extract the store name and index info from the schema.
+		const storesNames = Object.keys( schema );
+
+		if ( reset ) {
+			// An error can sometimes be thrown if we're deleting a DB that doesn't
+			// exist. Quickly opening and closing it avoids that issue.
+			const handle = await openDB( 'WXR-1.3' );
+			handle.close();
+
+			await deleteDB( 'WXR-1.3' );
+		}
+
+		this.db = await openDB( 'WXR-1.3', 1, {
+			upgrade: ( db ) => {
+				storesNames.forEach( ( storeName ) => {
+					const store = db.createObjectStore( storeName, {
+						keyPath: 'internalId',
+						autoIncrement: true,
+					} );
+
+					// If this store has indexes, add them to the store definition.
+					if ( schema[ storeName ].indexes ) {
+						schema[ storeName ].indexes.map( ( index ) =>
+							store.createIndex( index, index )
+						);
+					}
+				} );
+			},
+		} );
+	}
+
+	/**
+	 * Given a blob of data, run it against the schema for the given dataType, then store it
+	 * in the database.
+	 *
+	 * @param {string} dataType The type of data passed in the `data` blob.
+	 * @param {Object} data     The data to store.
+	 * @return {number} The autoincrement id of the data just stored.
+	 */
+	async storeData( dataType, data ) {
+		// If we don't know about this dataType, bail.
+		if ( ! schema[ dataType ] ) {
+			return;
+		}
+
+		if ( dataType === 'objects' ) {
+			return await this.storeObject( data.type, data.data );
+		}
+
+		const validatedData = {};
+
+		// Go over each field in the schema, and validate it.
+		schema[ dataType ].fields.forEach( ( field ) => {
+			// If it's an empty field, set an empty value, then move on.
+			if ( field.type === 'empty' ) {
+				validatedData[ field.name ] = '';
+				return;
+			}
+
+			// Skip any fields that have a filter which fails for this data blob.
+			if ( field.filter && ! field.filter( data ) ) {
+				return;
+			}
+
+			// If the field is read-only, get the default value.
+			if ( field.hasOwnProperty( 'writeable' ) && ! field.writeable ) {
+				validatedData[ field.name ] = field.default(
+					validatedData,
+					data
+				);
+				return;
+			}
+
+			let value;
+
+			// Check if we have a value for this field, or fall back to the default value.
+			if ( data.hasOwnProperty( field.name ) ) {
+				value = data[ field.name ];
+			} else if ( field.default ) {
+				value = field.default( validatedData, data );
+			}
+
+			if ( value === undefined ) {
+				return;
+			}
+
+			validatedData[ field.name ] = this.castValue( value, field.type );
+		} );
+
+		return await this.db.add( dataType, validatedData );
+	}
+
+	async storeObject( type, data ) {
+		if ( ! objectSchemas.hasOwnProperty( type ) ) {
+			return;
+		}
+
+		let validate = this.ajv.getSchema( type );
+		if ( ! validate ) {
+			this.ajv.addSchema( objectSchemas[ type ], type );
+			validate = this.ajv.getSchema( type );
+		}
+
+		if ( ! validate( data ) ) {
+			return;
```

Comment:
What would be a good way to report an error message here? At least we should do this:
```suggestion
			return false;
```